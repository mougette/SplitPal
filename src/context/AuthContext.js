import React, { useState} from "react";
import createDataContext from './createDataContext';
import {Post} from '../components/RestPost';
import {Get} from '../components/RestGet';

const authReducer = (state, action) => {
  switch (action.type) {
    case 'signout':
      return {token: null, email: ''};
    case 'signin':
    return {
            email: action.payload.email,
            token: action.payload.token,
          };
    case 'signup':
      return {
        token: action.payload.token,
        email: action.payload.email,
      };
  }
};

const signup = dispatch => {
    const [account, setAccount] = useState("");
  return ({email, password, password2, firstName, lastName, phoneNumber}) => {
    Post("https://wt9b6sq6k1.execute-api.us-east-2.amazonaws.com/Iteration_1/login",
                JSON.stringify({
                 email: email,
                 password: password,
                 password2: password2,
                    firstName: firstName,
                    lastName: lastName,
                    phoneNumber: phoneNumber,
                 }))
        .then(response => {
    console.log("Account: " + account + " Account End");
      if(response == "Account Successfully Made") {
              dispatch({
                  type: 'signup',
                  payload: {
                      token: 'some access token here',
                      email: email,
                  },
              });
      }
      else {
              alert(response)
      }
      }
      );


  };
};

const signin = dispatch => {
  return ({email, password}) => {
    // Do some API Request here
   Get("https://wt9b6sq6k1.execute-api.us-east-2.amazonaws.com/Iteration_1/login","?email="+email+"&password="+password)
   .then(response =>{
    if(response == "User info correct"){
    dispatch({
      type: 'signin',
      payload: {
        token: 'some access token here',
        email: email,
      },
    });
    }});
  };
};

const signout = dispatch => {
  return () => {
    dispatch({type: 'signout'});
  };
};

export const {Provider, Context} = createDataContext(
  authReducer,
  {signin, signout, signup},
  {token: null, email: ''},
);
