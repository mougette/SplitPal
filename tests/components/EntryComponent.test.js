import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react-native'
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {Provider as AuthProvider} from '../../src/context/AuthContext.js';
import {Context as AuthContext} from '../../src/context/AuthContext';
import {navigate} from '@reach/router'
import Entry, {buttonPress} from '../../src/components/EntryComponent';
import {NavigationContainer} from '@react-navigation/native';
import ReactTestUtils from 'react-dom/test-utils';
import { act } from 'react-dom/test-utils';
import {SafeAreaView, FlatList, Image, TouchableOpacity,Dimensions} from "react-native";

describe('Test Hooks', () => {
configure({adapter: new Adapter()});
    it('positive', () => {
    const tree = renderer
        .create(<Entry balance="20"/>);

    expect(tree).toMatchSnapshot();
    });

    it('pending', () => {
        const tree = renderer
            .create(<Entry balance="Pending"/>);

        expect(tree).toMatchSnapshot();
        });
    it('split', () => {
        const tree = renderer
            .create(<Entry balance="Split"/>);

        expect(tree).toMatchSnapshot();
        });
    it('click Button', async () => {
        buttonPress = jest.spyOn(Entry, buttonPress)
        const tree = shallow(<Entry balance="Pending"/>);
        console.log(tree.debug())
        const incButton = tree.find('ForwardRef');
        await act(async () => {
        incButton.simulate('click');
         });
        expect(buttonPress).toBeCalledTimes(1);
        });
});