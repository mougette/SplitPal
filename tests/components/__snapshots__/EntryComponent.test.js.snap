// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Hooks negative 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f5f5f5",
      "borderRadius": 7,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginVertical": 8,
      "padding": 10,
      "width": 400,
    }
  }
>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Object {
        "borderRadius": 30,
        "height": 60,
        "width": 60,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#bb2a2a",
        "borderRadius": 7,
        "borderWidth": 1,
        "color": "#2196F3",
        "fontSize": 20,
        "marginLeft": 10,
        "opacity": 1,
        "overflow": "hidden",
        "padding": 10,
        "width": "30%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 18,
        }
      }
    >
      -86
    </Text>
  </View>
</View>
`;

exports[`Test Hooks pending 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f5f5f5",
      "borderRadius": 7,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginVertical": 8,
      "padding": 10,
      "width": 400,
    }
  }
>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Object {
        "borderRadius": 30,
        "height": 60,
        "width": 60,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "grey",
        "borderRadius": 7,
        "borderWidth": 1,
        "color": "#2196F3",
        "fontSize": 20,
        "marginLeft": 10,
        "opacity": 1,
        "overflow": "hidden",
        "padding": 10,
        "width": "30%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 18,
        }
      }
    >
      Pending
    </Text>
  </View>
</View>
`;

exports[`Test Hooks positive 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f5f5f5",
      "borderRadius": 7,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginVertical": 8,
      "padding": 10,
      "width": 400,
    }
  }
>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Object {
        "borderRadius": 30,
        "height": 60,
        "width": 60,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2abb42",
        "borderRadius": 7,
        "borderWidth": 1,
        "color": "#2196F3",
        "fontSize": 20,
        "marginLeft": 10,
        "opacity": 1,
        "overflow": "hidden",
        "padding": 10,
        "width": "30%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 18,
        }
      }
    >
      20
    </Text>
  </View>
</View>
`;

exports[`Test Hooks split 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f5f5f5",
      "borderRadius": 7,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginVertical": 8,
      "padding": 10,
      "width": 400,
    }
  }
>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Object {
        "borderRadius": 30,
        "height": 60,
        "width": 60,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2a56a9",
        "borderRadius": 7,
        "borderWidth": 1,
        "color": "#2196F3",
        "fontSize": 20,
        "marginLeft": 10,
        "opacity": 1,
        "overflow": "hidden",
        "padding": 10,
        "width": "30%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 18,
        }
      }
    >
      Split
    </Text>
  </View>
</View>
`;
